include/zephyr/kernel.h

```diff
@@ -5416,6 +5416,18 @@ struct k_heap {
 void k_heap_init(struct k_heap *h, void *mem,
 		size_t bytes) __attribute_nonnull(1);
 
+/**
+ * @brief Free memory allocated by k_heap_alloc()
+ *
+ * Returns the specified memory block, which must have been returned
+ * from k_heap_alloc(), to the heap for use by other callers.  Passing
+ * a NULL block is legal, and has no effect.
+ *
+ * @param h Heap to which to return the memory
+ * @param mem A valid memory block, or NULL
+ */
+void k_heap_free(struct k_heap *h, void *mem) __attribute_nonnull(1);
```

Comment:
Not part of this PR, but: all but certain k_heap_free() allows NULL pointers and noops, as per the C malloc() behavior, and that we even have tests for this somewhere?  Note that the docs say NULL is allowed!  Some wires got crossed at some point.

Reply:
I'll add a commit to remoe the attribute.

Reply:
It turns out that the index is 1-based. Thus, `__attribute__((nonnull(1)))` refers to `h`, not `mem`. It's reasonable to require `h` to be non-NULL IMO, no?

Reply:
Checks for a non-null value are reasonable. Prepending 1 or 2 commits to update docs and possibly tweak some test cases would help.